### ISSUE TITLE
[Feature] Ignore components by import path / source file location

### ISSUE THREAD

**Issue by jmeistrich** [marklawlor/tailwindcss-react-native#16]
I'm trying to work around an issue of mixing reactjs components with react-native-web components. I need the babel transformer to skip the web components and only transform the react-native-web components. So for example I have a reactjs component `<Button>text</Button>` and it is being transformed because it starts with a capital letter. 

https://github.com/marklawlor/tailwindcss-react-native/issues/8 could be a workaround, so that any RN components would use a different prop and that would skip transforming `className` on the web components. But it would be ideal if our developers could use `div` and `View` interchangeably on our web app.

But I have another idea: a plugin option to set exclusion paths. So the babel transformer could look at the file's imports and if a JSX element was imported from an excluded path, skip processing the element. So if I exclude the `web` folder and `<Button .../>` was imported as `import Button from 'web/Button'` it would be skipped.

If this seems like a good idea to you I'd be happy to submit a PR. This is the one thing blocking me from being able to fully adopt it. Or is there an easier solution?

**Comment by jmeistrich** [comment]
An allow list would work for me just as well :). But the crucial point for the mixed use case is that it needs to control what's processed based on the path of the imported component, not control which files are processed. For example:

```
// tailwind-web.config.js
module.exports = {
  content: ["./native/*.{js,ts,jsx,tsx}"]
}

// index.jsx (mixes reactjs and react-native-web components)
import WebButton from './web/Button'
import NativeButton from './native/Button'

function App() {
    return (
        <div>
            <WebButton className="my-4">Web Button</WebButton>
            <NativeButton className="my-4">Native Button</NativeButton>
        </div>
    )
}
```

In this example it would be ideal if the allow list would run the transformer on `app.jsx` and transform `NativeButton` because it is allowed within the `native` folder but skip over `WebButton` because it is not.

**Comment by marklawlor** [comment]
I'm currently playing around with this proposal:

### Allow inline comments

`// tailwind-react-native: disable <ComponentName> <ComponentName>`

This will disable transforming for the specified components. Fail safe when no other options work.

### Add a `content` attribute to the babel config 

Skip files that do no match the `content` regex.

Only transform components that were declared locally or where imported from a path that matches the `content` glob.

This will resolve your scenario and should speed up the plugin.

### Add a `allowedImports` & `denyImports` babel options.

`allowedImports` defaults to `*`. Can also be an array of regex's
`deniedImports` defaults to `[]`

Components imported from a module must be included in the `allowedImports` and not included in the `denyImports`.

**Comment by marklawlor** [comment]
An initial version of this has been released in `0.0.24`. It will

* Only transform files that match the `content` globs.
* Only transform components from relative imports that match the `content` globs
* Only transform components imported from a node_module that match the `allowModules` globs
* Will not transform components imported from a node_module that match the `blockModules` globs.

There is not much documentation on this right now, I plan on creating a website that goes into more detail than the readme does.

I suspect there are many edge cases that are not covered, but this is a good start. Please create a new issue if you encounter any problems.

Lastly, I choose not to implement the inline comments, I'm not sure there is a good use-case for them.
